Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs

```diff
@@ -101,5 +116,77 @@ public TestsRepository(TrappistDbContext dbContext, IGlobalUtil util)
             await _dbContext.SaveChangesAsync();
         }
         #endregion
+        #region Test-Question-Selection
+        public async Task<List<QuestionAC>> GetAllTestCategoryQuestionsByIdAsync(int testId, int categoryId)
+        {
+            List<QuestionAC> questionListAC = new List<QuestionAC>();
+            var testQuestionList = await _dbContext.TestQuestion.Where(x=>x.TestId == testId).ToListAsync();
+            var questionList = await _dbContext.Question.Where(x => x.CategoryID == categoryId).Include(y => y.SingleMultipleAnswerQuestion).ThenInclude(x => x.SingleMultipleAnswerQuestionOption).ToListAsync();
+                     
+                 questionList.ForEach(question=> 
+                 { 
+                
+                    questionAc = new QuestionAC();
+                    questionAc.Question = Mapper.Map<Question, QuestionDetailAC>(question);
+                    questionAc.SingleMultipleAnswerQuestion = Mapper.Map<SingleMultipleAnswerQuestion, SingleMultipleAnswerQuestionAC>(question.SingleMultipleAnswerQuestion);
+                    if (testQuestionList.Exists(x=> x.QuestionId == questionAc.Question.Id && x.TestId == testId))
+                        questionAc.Question.IsSelect = true;
+                questionListAC.Add(questionAc);
+                });
+            return questionListAC;
+        }
+
+        public async Task<string> AddTestQuestionsAsync(List<QuestionAC> QuestionsToAddTest,int testId)
+        {
+            List<TestQuestion> testQuestionList = new List<TestQuestion>();
+            foreach (var questionToAdd in QuestionsToAddTest)
+            {
+                var question = Mapper.Map<QuestionDetailAC, Question>(questionToAdd.Question);
+                var TestCategory = _dbContext.TestCategory.FirstOrDefault(x=>x.CategoryId==question.CategoryID);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused 'TestCategory' local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1481)